torch/distributed/_tensor/placement_types.py

```diff
@@ -40,13 +40,15 @@ def _split_tensor(
         with_padding: bool = True,
         contiguous: bool = True,
     ) -> Tuple[List[torch.Tensor], List[int]]:
-        # NOTE: For with_padding option, we pad the tensor on each rank before calling
-        # the collectives (i.e. scatter/all_gather, etc.). This is because for gloo
-        # backend, it does not support uneven collectives, nccl supports some, but
-        # it might be slow compared to even size collective, we need to pad tensor
-        # before really calling the collective, and unpad/narrow it afterwards
-        # TODO: consider if we should remove this logic once ProcessGroupGloo
-        # support uneven list, and collective performance on par
```

Comment:
QQ regarding removing this comment: 
Are we removing the todo here because ProcessGroupGloo now support uneven list and collective performance is on par?

If so, do we want to remove the padding logic?

Reply:
no, there's nothing happened to processgroupgloo lol. The only thing is that I think we probably won't move to uneven sized collectives, even if processgroupgloo/nccl supports it well. The primary reason:
- we now use functional collectives, which we can't have uneven collective (small -> big tensor directly)
- I think uneven sized collective probably never going to be perf on par with even sized. 

So I think we should always keep the padding logic, and this TODO becomes useless.

Reply:
I see! Thanks for sharing all the context!